packages/store/src/actions/action-registry.ts

```diff
@@ -0,0 +1,117 @@
+import { inject, Injectable, OnDestroy, ɵisPromise } from '@angular/core';
+import {
+  defaultIfEmpty,
+  from,
+  isObservable,
+  mergeMap,
+  Observable,
+  of,
+  shareReplay,
+  takeUntil,
+  throwError
+} from 'rxjs';
+import { ɵActionHandlerMetaData } from '@ngxs/store/internals';
+
+import { ofActionDispatched } from '../operators/of-action';
+import { StateContextFactory } from '../internal/state-context-factory';
+import { ɵNgxsInternalDispatchedActions } from './dispatched-actions';
+
+// `action: Instance<ActionType>`.
+type ActionHandlerFn = (action: any) => void | Promise<void> | Observable<unknown>;
+
+interface InvokableActionHandlerMetaData extends ɵActionHandlerMetaData {
+  path: string;
+  instance: any;
+}
```

Comment:
This data structure should no longer be needed for the registry.
The registry should be super simple, only storing the handler functions, constructed and provided by the StateFactory.
The place in the StateFactory where the handler is created should contain enough access from its closure to be able to do all that it needs.